core/src/main/java/org/hisp/dhis/android/core/relationship/internal/RelationshipDownloadAndPersistCallFactory.kt

```diff
@@ -164,7 +164,7 @@ internal class RelationshipDownloadAndPersistCallFactory(
             corruptedRelationships.addAll(relationshipStore.getRelationshipsByItem(builder.build()))
         }
         for (r in corruptedRelationships) {
-            relationshipStore.deleteById(r)
+            r.uid()?.let { relationshipStore.deleteIfExists(it) }
```

Comment:
The `uid` is optional in the Relationship model, but it is actually optional. This is a legacy configuration from 2.29 where the Relationship objects didn't have a `uid`. Since 2.29 is not supported anymore, the relationship objects will always have a non-null `uid`.

We should clean up the Relationship model.